src/core_plugins/tests_bundle/index.js

```diff
@@ -13,9 +16,18 @@ export default (kibana) => {
     },
 
     uiExports: {
-      bundle: async (UiBundle, env, apps, plugins) => {
+      async __bundleProvider__(kbnServer) {
```

Comment:
Where is this used?  I see two other references to `__bundleProvider__` in this PR, and both of them are just exports.

Reply:
The other references define the `__bundleProvider__` export type, which is ultimately [defined here](https://github.com/elastic/kibana/blob/cc3fc83f4aa49f2bda99be6c5d5c914cfa86f0d3/src/ui/ui_exports/ui_export_types/webpack_customizations.js#L9). The export there is a reducer, which consumes all `__bundleProvider__` exports and writes them to the `uiBundleProviders` array in the uiExports object. This means that Kibana can access these providers at [`kbnServer.uiExports.uiBundleProviders`](http://github.com/elastic/kibana/blob/cc3fc83f4aa49f2bda99be6c5d5c914cfa86f0d3/src/ui/ui_bundles/ui_bundles_mixin.js#L7)

Reply:
Maybe a README in src/ui/ui_exports would be a good idea

Reply:
++ for a high-level readme in `src/ui/ui_exports`

Reply:
https://github.com/spalger/kibana/blob/implement/plugin-discovery/src/ui/ui_exports/README.md